pudb/shell.py

```diff
@@ -96,12 +96,19 @@ def have_ipython():
     # https://github.com/ipython/ipython/issues/9435
 
     try:
+        # Assume don't have IronPython unless this variable is
+        # defined, see:
+        # http://stackoverflow.com/questions/5376837/run-from-ipython
+        # this skips the pain of demandimporters when IPython
+        # is available but not actually active:
+        # https://bz.mercurial-scm.org/show_bug.cgi?id=5346
+        __IPYTHON__
```

Comment:
I don't understand how this can work. The first time IPython is imported is below this line. Wouldn't this only work if PuDB is started from within IPython? We want to support more than just that.